### ISSUE TITLE
sonobuoy: maybe we should delete the namespace

### ISSUE THREAD

**Issue by webern** [bottlerocket-os/bottlerocket-test-system#336]
It could be very easy for a user to end up with a target cluster that hasn't been cleaned up. In that case it is painful to see this:

```
[2022-02-28T02:49:29Z INFO  sonobuoy_test_agent] Initializing Sonobuoy test agent...
time="2022-02-28T02:49:29Z" level=error msg="Preflight checks failed"
time="2022-02-28T02:49:29Z" level=error msg="namespace already exists"
time="2022-02-28T02:49:29Z" level=info msg=deleted kind=namespace namespace=sonobuoy
time="2022-02-28T02:49:29Z" level=info msg=deleted kind=clusterrolebindings
time="2022-02-28T02:49:29Z" level=info msg=deleted kind=clusterroles
```

Maybe it should be a feature of the test agent that we automatically do a sonobuoy delete before we start the sonobuoy run.

**Comment by etungsten** [comment]
I do not think that's a safe behavior (even the current one). It could be that another process/person is running tests in the cluster. By deleting that namespace, it would nuke all their test progress. We should warn about the existing sonobuoy tests and error out and not delete the namespace in the test agent destructor. We rather want to  ensure deletion of the namespace after the full test has been done.

**Comment by webern** [comment]
Maybe, but do we anticipate multiple users stepping on each other? The problem as it stands is that having one conformance test run that errored out is arduous to fix and re-run.

**Comment by etungsten** [comment]
We do not, but it is always better to lean on the safer side when automating this since it is destructive if we just automatically delete something that was there before without confirming it's in use. I think it's precisely because we do not anticipate multiple processes/users testing in the same cluster at the same time that we do this. So that only out-of-band test runs that are manually initiated would require a manual `sonobuoy delete`. The automated process should ensure any test resources are deleted after its test run so this situation wouldn't happen.

I don't understand how it's hard to re-run if we do determine there's nothing happening with the already-deployed sonobuoy namespace. It's one `sonobuoy delete` call.

In the previous process, we failed fast after checking if the sonobuoy test namespace already exists and warn `"* Existing 'sonobuoy' namespace found in '${CLUSTER_NAME}'. Confirm it's not in use, then delete it before running this script."`. It would be nice to have that behavior here as well.

**Comment by webern** [comment]
> I don't understand how it's hard to re-run

It's not "hard", it's just un-ergonomic since the command to launch the test is currently 14 arguments and there's no facility for re-running a test by name.

The failure is about as fast as it can be now. The `sonobuoy run` command fails with a message about the namespace existing. There isn't really an earlier opportunity to check for the namespace unless the testsys cli did that, but that would be a big design shift.

**Comment by webern** [comment]
Closing this as undesirable. We will try to improve the ergonomics in other ways.
